lib/chef/provider/package/zypper.rb

```diff
@@ -97,7 +97,11 @@ def resolve_current_version(package_name)
 
         def resolve_available_version(package_name, new_version)
           search_string = new_version.nil? ? package_name : "#{package_name}=#{new_version}"
-          so = shell_out!("zypper", "--non-interactive", "search", "-s", "--provides", "--match-exact", "--type=package", search_string)
+          so = begin
+                shell_out!("zypper", "--non-interactive", "search", "-s", "--provides", "--match-exact", "--type=package", search_string)
+               rescue Exception => e
+                 Mixlib::ShellOut.new
+              end
```

Comment:
that just needs to be converted to `shell_out` instead of `shell_out!` which will probably fix it

Reply:
Thanks for the suggestion @lamont-granquist. I've made the changes.